src/tools/BaseTool.ts

```diff
@@ -9,6 +9,13 @@ import { GlobalStore } from "@/store";
  * - https://www.sitepoint.com/how-to-translate-from-dom-to-svg-coordinates-and-back-again/
  */
 export default abstract class BaseTool {
+  /**
+   * Used in ToolSelectNextPoint. Put in BaseTool due to a circular dependency:
```

Comment:
Sounds challenging -- you don't think there's another way to get around this?

Reply:
Would you mind speaking more to this? I'm getting an uneasy feeling from the mention of ToolSingleDot in this dependency chain, and the size of the chain; if there is a painful source of convoluted bugs/problems here, we should probably do whatever we need to to wrench this out.